scripts/zones/The_Garden_of_RuHmet/mobs/Ixaern_DRK.lua

```diff
@@ -15,25 +15,46 @@ require("scripts/globals/status")
 local entity = {}
 
 entity.onMobInitialize = function(IxAernDrkMob)
-    IxAernDrkMob:addListener("DEATH", "AERN_DEATH", function(mob)
+    IxAernDrkMob:addListener("DEATH", "AERN_DEATH", function(mob, killer)
         local timesReraised = mob:getLocalVar("AERN_RERAISES")
         if(math.random (1, 10) < 10) then
             -- reraise
             local target = mob:getTarget()
-            local targetid = 0
-            if target then
-                targetid = target:getTargID()
+            if
+                target:isPet() and
+                not target:isAlive()
+            then
+                target = target:getMaster()
             end
             mob:setMobMod(xi.mobMod.NO_DROPS, 1)
             mob:timer(9000, function(mobArg)
                 mobArg:setHP(mob:getMaxHP())
                 mobArg:setAnimationSub(3)
                 mobArg:resetAI()
                 mobArg:stun(3000)
-                local new_target = mobArg:getEntity(targetid)
-                if new_target and mobArg:checkDistance(new_target) < 40 then
-                    mobArg:updateClaim(new_target)
-                    mobArg:updateEnmity(new_target)
+                if
+                    mobArg:checkDistance(target) < 40 and
+                    target:isAlive()
+                then
+                    mobArg:updateClaim(target)
+                    mobArg:updateEnmity(target)
+                else
+                    local partySize = killer:getPartySize() -- Check for other available valid aggro targets
+                    local i = 1
+                    if killer ~= nil then
+                        for _, partyMember in pairs(killer:getAlliance()) do
+                            if partyMember:isAlive() and mobArg:checkDistance(partyMember) < 40 then
```

Comment:
Went and tested this on retail, a party member that never had enmity on the aern within 10~ units didn't get targeted and it didn't go after my SMN that killed it with a garuda melee attack. The SMN was over 25 units away.

Additionally, in a separate test I made sure my SMN had hate by flashing/meleeing down to low hp and letting Garuda kill the Aern with a melee attack. See log
[aern RR testing.txt](https://github.com/LandSandBoat/server/files/8886762/aern.RR.testing.txt)


tl;dr: change distance check to <25, and add a `-- TODO, change to check highest in enmity list`. There's no binding for the mob/player highest on the list _yet_